artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -20,33 +21,140 @@ impl Default for TimeArgs {
             hour: 0,
             minute: 0,
             second: 0,
-            nanoseconds: 0,
+            micros: 0,
         }
     }
 }
 
-pub fn as_time_args<T>(interp: &mut Artichoke, args: T) -> Result<TimeArgs, Error>
-where
-    T: IntoIterator<Item = Value>,
-{
-    let _interp = interp;
-    let _args = args;
-    Ok(TimeArgs::default())
+impl TimeArgs {
+    pub fn year(&self) -> Result<i32, Error> {
+        i32::try_from(self.year).map_err(|_| ArgumentError::with_message("year out of range").into())
+    }
+
+    pub fn month(&self) -> Result<u8, Error> {
+        // 1..=12 is gauranteed to fit into u8, so this cast is safe
+        match self.month {
+            #![allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+            1..=12 => Ok(self.month as u8),
+            _ => Err(ArgumentError::with_message("mon out of range").into()),
+        }
+    }
+
+    pub fn day(&self) -> Result<u8, Error> {
+        // 1..=31 is gauranteed to fit into u8, so this cast is safe
+        match self.day {
+            #![allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
```

Comment:
ditto above, using a `try_from` allows us to avoid this cast and clippy lint allows.